nomad/container.py

```diff
@@ -165,18 +173,26 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_slug}-{name}'.format(
+                project_slug=slugify(self.project_name), name=self.name)
+            container_id = self.id
+            self._lxd_name = '{prefix}-{id}'.format(
+                prefix=lxd_name_prefix[:63 - len(container_id)], id=container_id)
```

Comment:
How about only appending that ID when there's actually a name clash? What we're doing here is to burden our name with a hash for a situation that will rarely occur.